### PR TITLE
[CI] Add maintenance script to clean outdated vcpkg archives.

### PR DESCRIPTION
**Pull request by AenBleidd**
Small workflow clean-ups and fixes

Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -31,7 +31,9 @@ jobs:
       - name: Cache dependencies
         uses: actions/cache@v2.1.3
         with:
-          path: 3rdParty/buildCache
+          path: |
+            3rdParty/buildCache
+            !3rdParty/buildCache/android/vcpkgcache/vcpkg/archives/
```
Comment (talregev) [inline .github/workflows/android.yml:36]: Please change it to `!3rdParty/buildCache/android/vcpkgcache/`

File: deploy/s3.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                os.makedirs(os.path.join(dir, a[1]), exist_ok=True)
+                local_file = os.path.join(dir, a[1], a[2])
+                if (os.path.isfile(local_file)):
+                    print('Found local file', local_file)
+                    continue
+                s3_download(local_file, key, bucket)
+
+def download_all(dir, bucket):
+    l = s3_list(bucket)
+    if (l is not None):
```
Comment (talregev) [inline deploy/s3.py:77]: `if l:`

File: deploy/s3.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+def upload(os_name, dir, bucket, access_key, secret_key):
+    l = s3_list(bucket)
+    for path, _, files in os.walk(dir):
+        for name in files:
+            found = False
+            file_path = os.path.join(path, name)
+            dir_name = os.path.basename(path)
+            file_name = os_name + '-' + dir_name + '-' + name
+            if (l is not None):
```
Comment (talregev) [inline deploy/s3.py:52]: `if l:`

**Review by talregev — CHANGES_REQUESTED** [review]
Small changes are require

File: .github/workflows/android.yml (hunk 1 above)
Comment (AenBleidd) [reply .github/workflows/android.yml]: Fixed

File: deploy/s3.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                    key = k['Key']
+                    if (key == file_name):
+                        found = True
+                        break
+            if (not found):
+                s3_upload(file_path, file_name, bucket, access_key, secret_key)
+
+def download(os_name, dir, bucket):
+    l = s3_list(bucket)
+    if (l is not None):
```
[the remark this answers is not in the document]
Comment (AenBleidd) [reply deploy/s3.py]: Fixed

File: .github/workflows/linux.yml [hunk 5]
```diff
@@ -36,7 +36,9 @@ jobs:
       - name: Cache dependencies
         uses: actions/cache@v2.1.3
         with:
-          path: 3rdParty/buildCache
+          path: |
+            3rdParty/buildCache
+            !3rdParty/buildCache/linux/vcpkgcache/vcpkg/archives/
```
[the remark this answers is not in the document]
Comment (AenBleidd) [reply .github/workflows/linux.yml]: Fixed

File: deploy/s3.py (hunk 2 above)
Comment (AenBleidd) [reply deploy/s3.py]: Fixed

File: deploy/s3.py (hunk 3 above)
Comment (AenBleidd) [reply deploy/s3.py]: Fixed

**Review by talregev — APPROVED** [review]
I review the code.
CI is working.
I approve.

**Pull request merged**